plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbClient.java

```diff
@@ -0,0 +1,556 @@
+            return WriteMapping.sliceMapping(dataType, longDecimalWriteFunction(decimalType));
+        }
+        if (type instanceof CharType) {
+            return WriteMapping.sliceMapping("char(" + ((CharType) type).getLength() + ")", charWriteFunction());
+        }
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            String dataType;
+            if (varcharType.isUnbounded()) {
+                dataType = "longtext";
+            }
+            else if (varcharType.getBoundedLength() <= 255) {
+                dataType = "tinytext";
+            }
+            else if (varcharType.getBoundedLength() <= 65535) {
+                dataType = "text";
+            }
+            else if (varcharType.getBoundedLength() <= 16777215) {
+                dataType = "mediumtext";
+            }
+            else {
+                dataType = "longtext";
+            }
+            return WriteMapping.sliceMapping(dataType, varcharWriteFunction());
+        }
+        if (type == VARBINARY) {
+            return WriteMapping.sliceMapping("mediumblob", varbinaryWriteFunction());
+        }
+        if (type == DATE) {
+            return WriteMapping.longMapping("date", dateWriteFunction());
+        }
+        if (type instanceof TimeType) {
+            TimeType timeType = (TimeType) type;
+            if (timeType.getPrecision() <= MAX_SUPPORTED_DATE_TIME_PRECISION) {
+                return WriteMapping.longMapping(format("time(%s)", timeType.getPrecision()), timeWriteFunction(timeType.getPrecision()));
+            }
+            return WriteMapping.longMapping(format("time(%s)", MAX_SUPPORTED_DATE_TIME_PRECISION), timeWriteFunction(MAX_SUPPORTED_DATE_TIME_PRECISION));
+        }
+
+        throw new TrinoException(NOT_SUPPORTED, "Unsupported column type: " + type.getDisplayName());
+    }
+
+    @Override
+    public void createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        // MariaDB doesn't accept `some;column` in CTAS statements - so we explicitly block it and throw a proper error message
+        tableMetadata.getColumns().stream()
+                .map(ColumnMetadata::getName)
+                .filter(s -> s.contains(";"))
+                .findAny()
+                .ifPresent(illegalColumnName -> {
+                    throw new TrinoException(JDBC_ERROR, format("Incorrect column name '%s'", illegalColumnName));
+                });
+
+        super.createTable(session, tableMetadata);
+    }
+
+    @Override
+    public void renameColumn(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle jdbcColumn, String newColumnName)
+    {
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            String newRemoteColumnName = getIdentifierMapping().toRemoteColumnName(connection, newColumnName);
+            // MariaDB versions earlier than 10.5.2 do not support the RENAME COLUMN syntax
```

Comment:
Is there alternative syntax for older versions? Or is it similar to MySQL and requires providing list of ALL columns with their types?